lib/fiddle.rb

```diff
@@ -22,9 +21,11 @@ def self.last_error= error
 
     if WINDOWS
       class << self
-        alias win32_last_error last_error
+        def win32_last_error
+          FFI.errno.nonzero?
```

Comment:
```suggestion
          last_error.nonzero?
```

Reply:
I think it's better as-is to decouple each other and no accidentally break the other e.g. when changing `last_error`.
But if you prefer that feel free to apply it.

Maybe `win32_last_error`/`win32_last_socket_error` should never return `nil` either, but I don't know enough about Windows errors to make that call.

Reply:
If we want to decouple `last_error` and `win32_last_error`, we should stop `alias win32_last_error= last_error=` too.

I think that it's an overengineering without any real world problem but I'm OK with either of them:

```ruby
def win32_last_error
  last_error.nonzero?
end
alias win32_last_error= last_error=
```

```ruby
def win32_last_error
  FFI.errno.nonzero?
end
def win32_last_error=(errno)
  FFI.errno = error || 0
end
```

Reply:
OK, I did the second one (since they are supposed to be independent things but are not currently on JRuby).